client/src/main/java/org/kubeflow/client/model/TFReplica.java

```diff
@@ -49,64 +51,115 @@ public int getReplicas() {
   public TFReplica command(String command) {
     if (command != null && command.length() > 0) {
       String[] commands = command.split("\\s+");
-      this.spec.getTemplate().getSpec().getContainers().get(0).setCommand(Arrays.asList(commands));
+      for (V1Container container : this.spec.getTemplate().getSpec().getContainers()) {
+        container.setCommand(Arrays.asList(commands));
+      }
     }
     return this;
   }
 
+  public TFReplica env(String name, String value) {
+    for (V1Container container : this.spec.getTemplate().getSpec().getContainers()) {
+      if (container.getName().equals(JobConstants.KUBEFLOW_CONTAINER_NAME)) {
+        container.addEnvItem(new V1EnvVar().name(name).value(value));
+      }
+    }
+    return this;
+  }
+
+  public Map<String, String> getEnvs() {
+    for (V1Container container : this.spec.getTemplate().getSpec().getContainers()) {
+      if (container.getName().equals(JobConstants.KUBEFLOW_CONTAINER_NAME)) {
+        Map<String, String> envs = new HashMap<>();
+        for (V1EnvVar item : container.getEnv()) {
+          envs.put(item.getName(), item.getValue());
+        }
+        return envs;
+      }
+    }
+    return null;
+  }
+
   public String getCommand() {
-    return JobUtil.joinCommands(
-        this.spec.getTemplate().getSpec().getContainers().get(0).getCommand());
+    for (V1Container container : this.spec.getTemplate().getSpec().getContainers()) {
```

Comment:
create a common method findTFContainer for the tensorflow container, instead of duplicating the for loop everywhere

Reply:
fixed.